src/sink/probe.go

```diff
@@ -0,0 +1,39 @@
+package sink
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const backoff = 3 * time.Second
+
+var ErrProbeTimeout = errors.New("probe did not succeed")
+
+func WaitForEndpoint(endpoint string, timeout time.Duration) error {
```

Comment:
could we use the pester client already available with an increased number of retries for this instead of custom code?

Reply:
+1

Reply:
I tried to do so, but faced two blockers:
1. Using pester directly (e.g. not probing) does not fit very well with the problem we're trying to solve: Agent might take a few to start, but after that should respond promptly and with minimal retries
2. Pester does not easily allow to configure a global timeout, which we do want to do here.

The `for` loop is a bit custom, but I think it's pretty simple and (now) it's tested :)